libsolidity/formal/SMTEncoder.cpp

```diff
@@ -463,21 +463,25 @@ void SMTEncoder::endVisit(UnaryOperation const& _op)
 
 	createExpr(_op);
 
+	auto const* subExpr = &_op.subExpression();
```

Comment:
Why don't you turn this into a generic helper?

Reply:
This uses the helper `innermostTuple`, but yea, I guess it could take a pointer instead of a ref which forces the type. This way a few ifs could be avoided.